knowledge-base/listbox-filtering.md

```diff
@@ -0,0 +1,233 @@
+---
+title: Filter ListBox Data
+description: How to filter or search the data in a Telerik Blazor ListBox component programmatically.
+type: how-to
+page_title: How to Filter or Search ListBox Data
+slug: listbox-kb-filtering
+position: 
+tags: telerik, blazor, listbox, filter, search
+ticketid:
+res_type: kb
+---
+
+## Environment
+
+<table>
+    <tbody>
+        <tr>
+            <td>Product Version</td>
+            <td>4.6.0 and above</td>
+        </tr>
+        <tr>
+            <td>Product</td>
+            <td>Progress® Telerik® UI for Blazor</td>
+        </tr>
+    </tbody>
+</table>
+
+
+## Description
+
+This KB article answers the following questions:
+
+* How to enable users to search the ListBox component data?
+* How to implement programmatic ListBox data filtering?
+
+
+## Solution
+
+1. Implement a user interface that includes:
+    * A [`TextBox`]({%slug components/textbox/overview%}) for the search string;
+    * (optional) A selection component (DropDownButton, DropDownList, ContextMenu, SplitButton) for the filter operator;
+    * (optional) A [Button]({%slug components/button/overview%}) to clear the TextBox `Value`.
+    * (optional) A Button to start the filtering. Alternatively, use the [`ValueChanged` or `OnChange` event of the TextBox]({%slug components/textbox/events%}).
+1. Create a [`DataSourceRequest` object](/blazor-ui/api/Telerik.DataSource.DataSourceRequest) and populate its `Filters` property with a single [`FilterDescriptor`](/blazor-ui/api/Telerik.DataSource.FilterDescriptor).
+    * If you need more complex filtering logic, use one or more [`CompositeFilterDescriptor`](/blazor-ui/api/Telerik.DataSource.CompositeFilterDescriptor)s.
```

Comment:
You can use "one or more" with the singular form and avoid the strange "s" at the end.
```suggestion
    * If you need more complex filtering logic, use one or more [`CompositeFilterDescriptor`](/blazor-ui/api/Telerik.DataSource.CompositeFilterDescriptor).
```